packages/block-editor/src/components/global-styles/border-panel.js

```diff
@@ -22,10 +22,11 @@ import { getValueFromVariable, TOOLSPANEL_DROPDOWNMENU_PROPS } from './utils';
 import { overrideOrigins } from '../../store/get-block-settings';
 import { setImmutably } from '../../utils/object';
 import { getBorderPanelLabel } from '../../hooks/border';
-import { ShadowPopover } from './shadow-panel-components';
+import { ShadowPopover, getShadowPresets } from './shadow-panel-components';
 
 function useHasShadowControl( settings ) {
-	return !! settings?.shadow;
+	const shadows = getShadowPresets( settings );
+	return !! settings?.shadow && shadows.length > 0;
```

Comment:
Could not merge because they are two different variables `settings.shadow` and `shadows`. Moved the function below `useHasBorderWidthControl` as suggested.